homeassistant/components/traccar/device_tracker.py

```diff
@@ -171,17 +172,20 @@ def __init__(self, api, hass, async_see, scan_interval,
         events = await self._api.get_events(device_ids=device_ids,
                                             from_time=start_interval,
                                             to_time=end_interval,
-                                            event_types=self._event_types)
+                                            event_types=self._event_types.keys())
         if events is not None:
             for event in events:
-                device_name = list(filter(lambda dev:
-                                          dev.get('id') == event['deviceId'],
-                                          self._api._devices)
-                                   )[0].get('name')
-                self._hass.bus.fire('traccar_' + event["type"], {
-                    'device_traccar_id': event['deviceId'],
-                    'device_name': device_name,
-                    'type': event['type'],
-                    'serverTime': event['serverTime'],
-                    'attributes': event['attributes']
-                })
+                try:
+                    device_name = next((
+                        dev.get('name') for dev in self._api._devices
+                        if dev.get('id') == event['deviceId']))
+                except StopIteration:
+                    device_name = None
```

Comment:
Instead of catching `StopIteration`, put `None`, as the default value returned, as last argument to `next`.
```py
next((...), None)
```

Reply:
Done, thanks :)